src/binder/bind/bind_reading_clause.cpp

```diff
@@ -107,5 +112,38 @@ std::unique_ptr<BoundReadingClause> Binder::bindInQueryCall(const ReadingClause&
         tableFunctionDefinition->tableFunc, std::move(bindData), std::move(outputExpressions));
 }
 
+std::unique_ptr<BoundReadingClause> Binder::bindLoadFrom(
+    const parser::ReadingClause& readingClause) {
+    auto& loadFrom = reinterpret_cast<const LoadFrom&>(readingClause);
+    auto csvReaderConfig = bindParsingOptions(loadFrom.getParsingOptionsRef());
+    auto filePaths = bindFilePaths(loadFrom.getFilePaths());
+    auto fileType = bindFileType(filePaths);
+    auto readerConfig =
+        std::make_unique<ReaderConfig>(fileType, std::move(filePaths), std::move(csvReaderConfig));
+    if (readerConfig->fileType != FileType::CSV) {
+        throw BinderException("Load from non-csv file is not supported.");
+    }
+    if (readerConfig->getNumFiles() > 1) {
+        throw BinderException("Load from multiple csv files is not supported.");
+    }
+    auto csvReader = BufferedCSVReader(
+        readerConfig->filePaths[0], *readerConfig->csvReaderConfig, 0 /*expectedNumColumns*/);
+    csvReader.SniffCSV();
+    auto numColumns = csvReader.getNumColumnsDetected();
+    expression_vector columns;
+    auto stringType = LogicalType(LogicalTypeID::STRING);
+    for (auto i = 0; i < numColumns; ++i) {
+        auto columnName = "column" + std::to_string(i);
+        readerConfig->columnNames.push_back(columnName);
+        readerConfig->columnTypes.push_back(stringType.copy());
+        columns.push_back(createVariable(columnName, stringType));
+    }
+    auto offset = createVariable(std::string(Property::OFFSET_NAME), LogicalTypeID::INT64);
+    auto containsSerial = false;
+    auto info = std::make_unique<BoundFileScanInfo>(
```

Comment:
consider inherit a new class which doesn't require the offset and containsSerial